garak/_plugins.py

```diff
@@ -58,40 +62,105 @@ def _extract_modules_klasses(base_klass):
     def _load_plugin_cache(self):
         if not os.path.exists(self._plugin_cache_filename):
             self._build_plugin_cache()
```

Comment:
it looks like `_build_plugin_cache` currently leads to writes to the XDG cache dir, which means when `self._plugin_cache_filename` doesn't exist, the following action doesn't lead to anything that makes it exist in subsequent runs. would it make more sense for this `self._build_plugin_cache()` to be skipped and `update_user_file` be asserted True? is that operationally equivalent to the behavior here?

Reply:
I see your point, `self._plugin_cache_filename` is the package installed file and unless the install in corrupted the file must exist at this point.  It seems reasonable to convert this to an asset that reports the installed package as broken in some way.